core_lib/src/tool/eyedroppertool.cpp

```diff
@@ -143,10 +156,13 @@ void EyedropperTool::updateFrontColor()
 QColor EyedropperTool::getBitmapColor(LayerBitmap* layer)
 {
     BitmapImage* targetImage = layer->getLastBitmapImageAtFrame(mEditor->currentFrame(), 0);
-    if (targetImage == nullptr || !targetImage->contains(getLastPoint())) return QColor();
+    if (targetImage == nullptr || !targetImage->contains(getCurrentPoint())) return QColor();
 
     QColor pickedColour;
-    pickedColour.setRgba(qUnpremultiply(targetImage->pixel(getLastPoint().x(), getLastPoint().y())));
+    const QRgb pixelColor = targetImage->constScanLine(qFloor(qMin(getCurrentPoint().x(),getLastPoint().x())),
+                                                       qFloor(qMin(getCurrentPoint().y(),getLastPoint().y())));
+    pickedColour.setRgba(pixelColor);
+
```

Comment:
As discussed on Discord, the coordinates simply need to be floor()’d here when passed to pixel().